### PR TITLE
[release-4.16] OCPBUGS-44338: Prevent live migration process from flapping between step-2 and step-3

### PR DESCRIPTION
**Pull request by pliurh**
When a MCP is paused, the NetworkTypeMigrationTargetCNIInUse condition cannot become true. However NetworkTypeMigrationMTUReady will be false in that case, as the mtu-migration.service has been removed from unpaused MCPs. The migration will be flapping between step-2 and step-3. Checking the spec.defaultNetwork.Type before entering step-2 to ensure we don't go back from step-3.

### CODE REVIEW

File: pkg/controller/clusterconfig/migration.go [hunk 1]
```diff
@@ -111,7 +111,7 @@ func (r *ReconcileClusterConfig) prepareOperatorConfigForNetworkTypeMigration(ct
 		}
 		return nil
 	}
-	if !mtuApplied && cniReady {
+	if !mtuApplied && cniReady && operConfig.Spec.DefaultNetwork.Type != operv1.NetworkType(clusterConfig.Spec.NetworkType) {
 		if isRollback {
```
Comment (kyrtapz) [inline pkg/controller/clusterconfig/migration.go:115]: Should this if block happen after mtuMigration retrieval? Currently if calculateRoutableMTU fails we will not re-enter the step 2 block on rollback.

Comment (pliurh) [reply pkg/controller/clusterconfig/migration.go]: Good point. Fixed.

**Pull request merged**